### PR TITLE
Added Testing for components

### PR DESCRIPTION
**Pull request by Talha-Hanif5666**
### Highlights ✔

- [ ] Added test functions for logic files.
- [ ] Added test functions for all components.

### CODE REVIEW

File: src/logic/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import operate from './operate';
+
+describe('Addition', () => {
+  test('should Multiply 4 and 7 tobe equal 28', () => {
+    const result = operate(4, 7, 'x');
+    expect(result).toBe('28');
+  });
+});
```
Comment (roshan-bajgain) [inline src/logic/operate.test.js]: - [ ] I would suggest you please add more tests in your operate.test.js as there is only one test that is `Addition`, So I would suggest please test all the functionalities `-, *, ÷, %`

File: src/logic/calulate.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+describe('Button 0', () => {
+  test('display 0 when button zero is pressed', () => {
+    const op = { next: '0' };
+    const expected = {};
+    const buttonName = '0';
+    const result = calculate(op, buttonName);
+    expect(result).toEqual(expected);
+  });
+});
```
Comment (roshan-bajgain) [inline src/logic/calulate.test.js]: - [ ]  I would suggest you please add more tests in your calculator.test.js as all the functionalities and scenarios should be tested.

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Talha-Hanif5666 and @aubinleyoung, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi @Talha-Hanif5666 and @aubinleyoung, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job adding all the required test cases :rocket: awesome


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**